library/Vanilla/Embeds/GettyEmbed.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0
+ */
+
+namespace Vanilla\Embeds;
+
+use Exception;
+/**
+ * Getty Embed.
+ */
+class GettyEmbed extends Embed {
+
+    /** @inheritdoc */
+    protected $domains = ['gettyimages.ca', 'gty.im', 'gettyimages.com'];
+
+    /**
+     * GettyEmbed constructor.
+     */
+    public function __construct() {
+        parent::__construct('getty', 'image');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function matchUrl(string $url) {
+        $data = [];
+
+        if ($this->isNetworkEnabled()) {
+            preg_match(
+                '/https?:\/\/www\.gettyimages\.(com|ca).*?(?<postID>\d+)(?!.*\d)/i',
+                $url,
+                $post
+            );
+            if (!$post) {
+                throw new Exception('Unable to get post ID.', 400);
+            }
+
+            $oembedData = $this->oembed("http://embed.gettyimages.com/oembed?url=http://gty.im/".$post['postID']);
+
+            if ($oembedData) {
+                $data = $oembedData;
+            }
+
+            if (array_key_exists('html', $data)) {
+                $data['attributes'] = $this->parseResponseHtml($data['html']);
+            }
+
+            if ($post) {
+                $data['attributes']['postID'] = $post['postID'];
+            }
+        }
+        return $data;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function renderData(array $data): string {
+        $url = "//www.gettyimages.com/detail/".$data['attributes']['postID'];
+        $encodedURL = htmlspecialchars($url);
+        $encodedData = json_encode($data);
+        $sanitizedData = htmlspecialchars($encodedData);
+        $encodeID = htmlspecialchars($data['attributes']['id']);
+
+        $result = <<<HTML
+<a id="{$encodeID}" data-json={$sanitizedData} class='gie-single js-gettyEmbed' href="{$encodedURL}">Embed from Getty Images</a>
```

Comment:
I'm not a big fan of blindly stuffing the `data-json` attribute, if we can help it. I think we only need eight elements, so it's not making sense to me why we would include almost twice that many. Could you use several specific `data-*` attributes, instead of one giant monolithic one?

Reply:
The `data-json` attribute should probably be quoted.

Reply:
Let's stick with a consistent quote style. The `class` attribute is using "single quotes", while everything else is double.